climate_categories/tests/test_conversions.py

```diff
@@ -157,95 +158,84 @@ def test_good_csv(self, tmp_path):
-            csv_line_number=9,
+            csv_line_number=4,
             csv_original_text='b + "argl.5" + c,,,D,nobody needs argl',
         )
         assert conv.rule_specs[3] == conversions.ConversionRuleSpec(
             factors_categories_a={"b": 1, "argl,5": 1, "c": 1},
             factors_categories_b={"D": 1},
             auxiliary_categories={"aux1": set(), "aux2": set()},
-            csv_line_number=10,
+            csv_line_number=5,
             csv_original_text='b + "argl,5" + c,,,D',
         )
         assert conv.rule_specs[4] == conversions.ConversionRuleSpec(
             factors_categories_a={"b": 1, 'argl"5': 1, "c": 1},
             factors_categories_b={"D": 1},
             auxiliary_categories={"aux1": set(), "aux2": set()},
-            csv_line_number=11,
+            csv_line_number=6,
             csv_original_text='b + "argl\\"5" + c,,,D',
         )
         assert conv.rule_specs[5] == conversions.ConversionRuleSpec(
             factors_categories_a={"b": 1, 'argl"5': 1, "c": 1},
             factors_categories_b={"D": 1, "E": 1},
             auxiliary_categories={"aux1": set(), "aux2": set()},
-            csv_line_number=12,
+            csv_line_number=7,
             csv_original_text='b + "argl\\"5" + c,,,D+E',
         )
         assert conv.rule_specs[6] == conversions.ConversionRuleSpec(
             factors_categories_a={"argl,5": 1},
             factors_categories_b={"D": 1},
             auxiliary_categories={"aux1": set(), "aux2": set()},
-            csv_line_number=13,
+            csv_line_number=8,
             csv_original_text='"argl,5",,,D',
         )
         assert conv.rule_specs[7] == conversions.ConversionRuleSpec(
             factors_categories_a={"+": 1, "-": -1},
             factors_categories_b={"-": 1},
             auxiliary_categories={"aux1": set(), "aux2": set()},
-            csv_line_number=14,
+            csv_line_number=9,
             csv_original_text='"+" - "-",,,"-"',
         )
         assert conv.rule_specs[8] == conversions.ConversionRuleSpec(
             factors_categories_a={"b": 1},
             factors_categories_b={"asdf": 1, "4": 1},
             auxiliary_categories={"aux1": set(), "aux2": set()},
-            csv_line_number=15,
+            csv_line_number=10,
             csv_original_text="b,,,asdf+4",
         )
 
         assert repr(conv) == "<ConversionSpec 'A' <-> 'B' with 9 rules>"
 
     def test_formula_broken(self):
-        csv = ["comment,no comment", "", "A,B,comment", "A.1+,C,broken formula"]
-        with pytest.raises(ValueError, match="line 4.*Could not parse"):
-            climate_categories._conversions.ConversionSpec.from_csv(csv)
-
-    def test_meta_data_incomplete(self):
-        csv = ["comment,no comment", "references"]
-        with pytest.raises(
-            ValueError, match="Meta data specification is incomplete in line 2"
-        ):
-            climate_categories._conversions.ConversionSpec.from_csv(csv)
```

Comment:
We could still test if there is a useful error if the YAML is incomplete, e.g.:

```
# comment: no comment
# references
data,data
```
should lead to a somewhat legible error, I think.

Reply:
If you miss the colon, one gets `strictyaml.exceptions.YAMLValidationError` from strictyaml, if the key is empty, it gets read as an empty string. Not sure if it's necessary to test this here as well.

Reply:
Sounds good enough.